### PR TITLE
Time elapsed parser

### PR DESCRIPTION
**Pull request by j-mai**
This PR:

- Implements the time-elapsed parser and adds configuration for it in `glaukos.yaml`.
- Changes the `Parser` interface by adding a `Name` function and making it so that `Parse` no longer returns an error.

### CODE REVIEW

File: eventmetrics/parsers/eventConfig.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+// If an EventInfo's validator is nil, it means there is no validation needed, so the event
+// is valid by default.
+func (e EventInfo) Valid(event interpreter.Event) (bool, error) {
+	if e.Validator == nil {
+		return true, nil
+	}
+	return e.Validator.Valid(event)
+}
+
+// TimeLocation is an enum to determine which timestamp should be used in timeElapsed calculations
```
Comment (kristinapathak) [inline eventmetrics/parsers/eventConfig.go:93]: `TimeLocation` stuff should be its own file.

File: eventmetrics/parsers/eventConfig.go [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	}
+
+	if bootTime, err := e.BootTime(); err == nil {
+		return time.Unix(bootTime, 0)
+	} else {
+		return time.Time{}
+	}
+}
+
+// SessionType is an enum to determine which session an event should be searched from
```
Comment (kristinapathak) [inline eventmetrics/parsers/eventConfig.go:135]: SessionType should be its own file.

File: eventmetrics/parsers/timeElapsedParser.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	"github.com/xmidt-org/interpreter"
+	"github.com/xmidt-org/interpreter/history"
+	"github.com/xmidt-org/themis/xlog"
+)
+
+const (
+	firmwareLabel     = "firmware"
+	hardwareLabel     = "hardware"
+	rebootReasonLabel = "reboot_reason"
+	errNoFwHwLabel    = "err_no_firmware_or_hardware"
```
Comment (kristinapathak) [inline eventmetrics/parsers/timeElapsedParser.go:39]:
```suggestion
	noFwHwReason    = "err_no_firmware_or_hardware"
```

File: eventmetrics/parsers/timeElapsedParser.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+		measures:      measures,
+		name:          config.Name,
+	}, nil
+}
+
+// Parse implements the Parser interface. Calculates the time-elapsed and then adds to prometheus metrics if possible.
+func (t *TimeElapsedParser) Parse(event interpreter.Event) {
+	restartTime, err := t.calculateTimeElapsed(event)
+	if err != nil || restartTime <= 0 {
+		return
```
Comment (kristinapathak) [inline eventmetrics/parsers/timeElapsedParser.go:129]: Do we want this case to be counted for the unparseable events metric?

**Review by kristinapathak — COMMENTED** [review]
looks good! Had a few suggestions/questions.

File: eventmetrics/parsers/timeElapsedParser.go (hunk 4 above)
Comment (j-mai) [reply eventmetrics/parsers/timeElapsedParser.go:139]: Good point! Would be helpful for monitoring/debugging any problems.

**Review by kristinapathak — APPROVED** [review]
looks great!

**Pull request merged**